rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -244,6 +269,37 @@ __check_parameter_value_in_range(
     result.reason = format_range_reason(descriptor.name, "floating point");
     return result;
   }
+
+  if (!descriptor.floating_point_range.empty() &&
+    value.get_type() == rclcpp::PARAMETER_DOUBLE_ARRAY)
+  {
+    std::vector<double> v = value.get<std::vector<double>>();
+    auto fp_range = descriptor.floating_point_range.at(0);
+    for (const double & val : v) {
+      if (__are_doubles_equal(val, fp_range.from_value) || __are_doubles_equal(val,
```

Comment:
This is more or less C&P of the function above, please refactor, to use common code in both cases.

Reply:
Done